model/src/main/kotlin/OrtResult.kt

```diff
@@ -157,6 +158,47 @@ data class OrtResult(
         return analyzerErrors.zipWithDefault(scannerErrors, emptySet()) { left, right -> left + right }
     }
 
+    /**
+     * Return a map of license findings for each project or package [Identifier]. The license findings for projects are
+     * mapped to a list of [PathExclude]s matching the locations where a license was found. This list is only populated
+     * if all file locations are excluded. The list is empty for all dependency packages, as path excludes are only
+     * applied to the projects.
+     *
+     * If [omitExcluded] is set to true, excluded projects / packages are omitted from the result.
+     */
+    fun collectLicenseFindings(omitExcluded: Boolean = false) =
+            sortedMapOf<Identifier, MutableMap<LicenseFinding, List<PathExclude>>>().also { findings ->
+                val excludes = repository.config.excludes
+
+                scanner?.results?.scanResults?.forEach { result ->
+                    val project = analyzer?.result?.projects?.find { it.id == result.id }
+
+                    if (!omitExcluded || excludes?.isPackageExcluded(result.id, this) != true) {
+                        result.results.flatMap { it.summary.licenseFindings }.forEach { finding ->
+                            val matchingExcludes = mutableSetOf<PathExclude>()
+                            val isExcluded = finding.locations.all { location ->
+                                excludes != null && excludes.paths.any { exclude ->
+                                    exclude.matches(location.path).also { matches ->
+                                        if (matches) matchingExcludes += exclude
+                                    }
+                                }
+                            }
+
+                            // TODO: Also filter copyrights excluded by path excludes.
+
+                            // Add the matching excludes only for projects and if all file locations where the license
+                            // was found are excluded.
+                            findings.getOrPut(result.id) { mutableMapOf() }[finding] =
+                                    if (project != null && isExcluded) {
```

Comment:
So if `project == null` we do not care about the excludes. Which means it makes sense to not even determine them in that case. So how about removing this check here and instead after line 174 doing something like

```
if (project == null) {
    findings[result.id] = emptyList()
} else if (excludes?.isPackageExcluded(result.id, this) != true) {
    // ...
```

Reply:
The values of the `findings` map need to be maps of type `Map<LicenseFinding, List<PathExclude>`, setting this separately would duplicate code. I have moved the check for `project != null` to a different place to avoid checking path excludes for packages.